### PR TITLE
SVC-23829 Add local user svna

### PR DESCRIPTION
**Pull request by bsper2**
Also PDK update and fixes related to that

### CODE REVIEW

**Comment by bsper2** [comment]
Tested on idds-test

File: manifests/ssh.pp [hunk 1]
```diff
@@ -21,33 +21,31 @@
...
-  $dev_users,
-  $dev_groups,
-){
-
+  Array[String] $acctd_subnets,
+  Array[String] $acctd_groups,
+  Array[String] $acctd_users,
+  Array[String] $dev_subnets,
+  Array[String] $dev_users,
+  Array[String] $dev_groups,
```
Comment (billglick) [inline manifests/ssh.pp:29]: Did `pdk validate -a` automatically typecast these variable?

File: manifests/users.pp [hunk 2]
```diff
@@ -56,6 +55,17 @@
...
+  user { 'svna':
+    ensure         => 'present',
+    name           => 'svna',
+    comment        => 'NCSA svna',
+    home           => '/home/svna',
+    managehome     => 'true',
+    password       => '!!',
+    purge_ssh_keys => 'false',
+    shell          => '/bin/bash',
+    system         => 'false',
```
Comment (billglick) [inline manifests/users.pp]: Are you sure you don't want this to be a `system` account like most of the other users we are adding? I fear if it isn't there is a chance it collides with an LDAP user UID.

**Review by billglick — APPROVED** [review]
Generally looks good. But did have a couple of questions.

File: manifests/ssh.pp (hunk 1 above)
Comment (bsper2) [reply manifests/ssh.pp:29]: No, pdk validate complained about it but a `-a` didn't fix it. Had to manually add them.

Are you thinking they should be a different type?

File: manifests/users.pp (hunk 2 above)
Comment (bsper2) [reply manifests/users.pp]: Good point. I just grabbed the stanza for `amie` user so just used the same value. I updated `amie` to be `system` as well while I'm at it.

File: manifests/ssh.pp (hunk 1 above)
Comment (billglick) [reply manifests/ssh.pp:29]: I was just curious. No change needed.

File: manifests/users.pp (hunk 2 above)
Comment (billglick) [reply manifests/users.pp]: I'm not sure changing the Amie user is a good idea. Wouldn't that change the UID and possibly make some data unavailable?

Comment (bsper2) [reply manifests/users.pp]: From the docs it says flipping `system` won't trigger it to change the uid if the user already exists. Also tested that on `idds-test`  where `svna` was already created as a non system user and it didn't try and change anything.

**Pull request merged**